lib/HalDependencyResolver.js

```diff
@@ -520,7 +545,34 @@ HalDependencyResolver.prototype = {
 		return result;
 	},
 
+	_getModuleFirstDependecy: function(modules, currentModule) {
+		if (!currentModule) {
+			return when.reject();
+		}
+		// Check if it also has dependencies
+		if (currentModule.dependencies.length === 0) {
+			return when.resolve(currentModule);
+		} else {
+			var missingModule = null;
+
+			currentModule.dependencies.map(function(missing){
+					modules.map(function(module){
+						if (module.version === missing.version &&
+								module.func === missing.func &&
+								module.name === missing.name) {
+
+								missingModule = module;
```

Comment:
this is kinda weird, should we be breaking out of this iteration?  are we potentially overwriting this?